### PR TITLE
cloned manifold_learner

### PR DESCRIPTION
**Pull request by vortex-17**
Hey this is the branch which has got the update.
Thanks a lot

### CODE REVIEW

File: pyloras/_loras.py [hunk 1]
```diff
@@ -128,8 +133,7 @@ def _initialize_params(self, X, y, rng):
         self.n_affine_ = f_size if self.n_affine is None else self.n_affine
 
         if self.manifold_learner:
-            self._check_2d_manifold_learner()
-            self.manifold_learner_ = self.manifold_learner
+            self.manifold_learner_ = self._check_2d_manifold_learner()
```
Comment (zoj613) [inline pyloras/_loras.py:135]: You forgot to add the `clone` function inside of `_check_2d_manifold_learner` and return it as the method's return value.

**Comment by zoj613** [comment]
you're doing good @vortex-17 , just have to add the missing `clone` functionality inside the `_check_2d_manifold_learner` method and return the cloned object as its return value.

File: pyloras/_loras.py (hunk 1 above)
Comment (vortex-17) [reply pyloras/_loras.py:135]: should it be 
```
def _check_2d_manifold_learner(self):
        if (not hasattr(self.manifold_learner, "fit_transform") or
                not hasattr(self.manifold_learner, "set_params")):
            raise ValueError(
                "The 2d manifold learner must implement the ``fit_transform`` "
                "and ``set_params`` methods"
            )
        return clone(self.manifold_learner)
```
??

Comment (zoj613) [reply pyloras/_loras.py:135]: Yes, that is correct. Dont forget to import the clone function

Comment (vortex-17) [reply pyloras/_loras.py:135]: Alright. I have pushed the changes. 
Thanks a ton for helping me out with everything.

**Comment by zoj613** [comment]
Great, now lets see if the changes pass the CI test suite.

**Comment by zoj613** [comment]
just merged your changes, thanks @vortex-17. More PR's are always welcome.

**Comment by vortex-17** [comment]
Thank you for helping me out with this PR. will look forward for more collabs and PRs.

**Pull request merged**